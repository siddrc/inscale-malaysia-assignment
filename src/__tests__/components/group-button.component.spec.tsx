import React from 'react';
import {
    render,
    cleanup,
    RenderResult,
    fireEvent
  } from '@testing-library/react';
import {GroupButtonComponent,GroupButtonComponentProps, Button} from '../../components/group-button.component';
describe(`Group button unit tests`,()=>{
    let component:RenderResult;
    let buttons:Array<Button> = new Array<Button>();
    let onClickHandler = ()=>{}
    let groupButtonComponentProps:GroupButtonComponentProps = {buttons,onClickHandler} 
    beforeEach(async ()=>{
        component = await render(<GroupButtonComponent {...groupButtonComponentProps}/>);
    })
    afterEach(()=>{
        cleanup();
        groupButtonComponentProps = {buttons:[],onClickHandler:()=>{}} 
    });
    it('to render it must take an array of button data',()=>{
        const {queryAllByTestId} = component;
        expect(queryAllByTestId('group-button').length).toEqual(0);
    });
    it('if "x" number of buttons are passed,it must render those many buttons only',()=>{
        let buttons:Array<Button> = new Array<Button>();
        buttons.push({label:'test',value:'test'}) 
        groupButtonComponentProps =  {buttons,onClickHandler:()=>{}} ;
        component = render(<GroupButtonComponent {...groupButtonComponentProps}/>);
        const {queryAllByTestId} = component;
        expect(queryAllByTestId('group-button').length).toEqual(1);
    });
    it('if "x" number of buttons are passed,it must render those many buttons only',()=>{
        let buttons:Array<Button> = new Array<Button>();
        buttons.push({label:'test',value:'test'}) 
        groupButtonComponentProps =  {buttons,onClickHandler:()=>{}} ;
        component = render(<GroupButtonComponent {...groupButtonComponentProps}/>);
        expect(component).toMatchSnapshot();
    });
    it('the button label must match the buttons array passed',()=>{
        let buttons:Array<Button> = new Array<Button>();
        buttons.push({label:'testLabel',value:'test'},{label:'testButtonLabel-1',value:'testButtonValue'}) 
        groupButtonComponentProps =  {buttons,onClickHandler:()=>{}} ;
        component = render(<GroupButtonComponent {...groupButtonComponentProps} />);
        const {getByText} = component;
        expect(getByText('testLabel')).toBeTruthy();
    });
    it('the button label must match the buttons array passed-snapshot test',()=>{
        let buttons:Array<Button> = new Array<Button>();
        buttons.push({label:'testLabel',value:'test'},{label:'testButtonLabel-1',value:'testButtonValue'}) 
        groupButtonComponentProps =  {buttons,onClickHandler:()=>{}} ;
        component = render(<GroupButtonComponent {...groupButtonComponentProps} />);
        expect(component).toMatchSnapshot();
    });
    it('the function passed in props must be called when any of the button in group is clicked',done=>{
        let someOnClickHandler=()=>done();
        const buttons:Array<Button> = new Array<Button>();
        buttons.push({label:'testLabel',value:'test'})
        groupButtonComponentProps =  {buttons,onClickHandler:()=>someOnClickHandler()} ;
        component = render(<GroupButtonComponent {...groupButtonComponentProps} />);
        const {getByText} = component;
        const node = getByText('testLabel');
        fireEvent.click(node);
    })
})