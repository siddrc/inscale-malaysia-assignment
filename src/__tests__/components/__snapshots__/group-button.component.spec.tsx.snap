// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Group button unit tests if "x" number of buttons are passed,it must render those many buttons only 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="btn-group"
        role="group"
      />
    </div>
    <div>
      <div
        class="btn-group"
        role="group"
      >
        <button
          class="btn btn-primary"
          data-testid="group-button"
          type="button"
          value="test"
        >
          test
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="btn-group"
      role="group"
    >
      <button
        class="btn btn-primary"
        data-testid="group-button"
        type="button"
        value="test"
      >
        test
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Group button unit tests the button label must match the buttons array passed-snapshot test 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="btn-group"
        role="group"
      />
    </div>
    <div>
      <div
        class="btn-group"
        role="group"
      >
        <button
          class="btn btn-primary"
          data-testid="group-button"
          type="button"
          value="test"
        >
          testLabel
        </button>
        <button
          class="btn btn-primary"
          data-testid="group-button"
          type="button"
          value="testButtonValue"
        >
          testButtonLabel-1
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="btn-group"
      role="group"
    >
      <button
        class="btn btn-primary"
        data-testid="group-button"
        type="button"
        value="test"
      >
        testLabel
      </button>
      <button
        class="btn btn-primary"
        data-testid="group-button"
        type="button"
        value="testButtonValue"
      >
        testButtonLabel-1
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
